pyscf/pbc/cc/eom_kccsd_ghf.py

```diff
@@ -1279,6 +1280,551 @@ def make_imds(self, eris=None):
+        Hr2[ki, kj, ka] += tmp_ij
+        Hr2[kj, ki, ka] -= tmp_ij.transpose(1, 0, 2, 3)
+
+    vector = amplitudes_to_vector_ee(Hr1, Hr2, kshift, kconserv_r2)
+    return vector
+
+
+def eeccsd_diag(eom, kshift, imds=None):
+    if imds is None: imds = eom.make_imds()
+    t1, t2 = imds.t1, imds.t2
+    nkpts, nocc, nvir = t1.shape
+    kconserv = eom.kconserv
+    kconserv_r1 = eom.get_kconserv_ee_r1(kshift)
+    kconserv_r2 = eom.get_kconserv_ee_r2(kshift)
+
+    Hr1 = np.zeros((nkpts, nocc, nvir), dtype=t1.dtype)
+    for ki in range(nkpts):
+        ka = kconserv_r1[ki]
+        Hr1[ki] -= imds.Foo[ki].diagonal()[:, None]
+        Hr1[ki] += imds.Fvv[ka].diagonal()[None, :]
+        Hr1[ki] += np.einsum('iaai->ia', imds.Wovvo[ki, ka, ka])
+
+    Hr2 = np.zeros((nkpts, nkpts, nkpts, nocc, nocc, nvir, nvir),
+                   dtype=t1.dtype)
+    # TODO allow partition='mp'
+    if eom.partition == "mp":
+        raise NotImplementedError
+    else:
+        for ki, kj, ka in kpts_helper.loop_kkk(nkpts):
+            kb = kconserv_r2[ki, ka, kj]
+            Hr2[ki, kj, ka] -= imds.Foo[ki].diagonal()[:, None, None, None]
+            Hr2[ki, kj, ka] -= imds.Foo[kj].diagonal()[None, :, None, None]
+            Hr2[ki, kj, ka] += imds.Fvv[ka].diagonal()[None, None, :, None]
+            Hr2[ki, kj, ka] += imds.Fvv[kb].diagonal()[None, None, None, :]
+
+            Hr2[ki, kj, ka] += np.einsum('jbbj->jb', imds.Wovvo[kj, kb, kb])[None, :, None, :]
+            Hr2[ki, kj, ka] += np.einsum('ibbi->ib', imds.Wovvo[ki, kb, kb])[:, None, None, :]
+            Hr2[ki, kj, ka] += np.einsum('jaaj->ja', imds.Wovvo[kj, ka, ka])[None, :, :, None]
+            Hr2[ki, kj, ka] += np.einsum('iaai->ia', imds.Wovvo[ki, ka, ka])[:, None, :, None]
+
+            Hr2[ki, kj, ka] += np.einsum('ijij->ij', imds.Woooo[ki, kj, ki])[:, :, None, None]
+            Hr2[ki, kj, ka] += np.einsum('abab->ab', imds.Wvvvv[ka, kb, ka])[None, None, :, :]
+
+            # This is to make t2 are non-zero
+            # Note that `kconserv` is used instead of `kconserv_r2`
+            kk = kconserv[ka, kj, kb]
+            Hr2[ki, kj, ka] -= np.einsum('kjab,kjab->jab', imds.Woovv[kk, kj, ka], imds.t2[kk, kj, ka])[None, :, :, :]
+            kk = kconserv[ka, ki, kb]
+            Hr2[ki, kj, ka] -= np.einsum('kiab,kiab->iab', imds.Woovv[kk, ki, ka], imds.t2[kk, ka, ka])[:, None, :, :]
+
+            kc = kconserv[ki, kb, kj]
+            Hr2[ki, kj, ka] -= np.einsum('ijcb,ijcb->ijb', imds.Woovv[ki, kj, kc], imds.t2[ki, kj, kc])[:, :, None, :]
+            kc = kconserv[ki, ka, kj]
+            Hr2[ki, kj, ka] -= np.einsum('ijca,ijca->ija', imds.Woovv[ki, kj, kc], imds.t2[ki, kj, kc])[:, :, :, None]
+
+    # Make sure 4th argument you pass is `kconserv_r2`
+    vector = amplitudes_to_vector_ee(Hr1, Hr2, kshift, kconserv_r2)
+    return vector
+
+
+# TODO complete this method
+def mask_frozen_ee(eom, vector, kshift, const=LARGE_DENOM):
```

Comment:
Remove it if you are not going to complete it. There is a `TODO` reminder up there in `kernel_ee()` so this one is not needed.